src/components/chat.tsx

```diff
@@ -0,0 +1,53 @@
+// @ts-nocheck
+// import type { NextPage } from 'next';
+// import { PropsWithChildren } from 'react';
+import Image from 'next/image';
+import avatarReceiver from '../images/avatar/male.png';
+import avatarSender from '../images/avatar/female.png';
+
+// NextJs requires more explicit typing for props when defining a `NextPage`
+const Chat = (props) => {
```

Comment:
We are assuming that every message will be replied. In other words, every time the sender sends a message, there will always be a reply back. We don't want that assumption because it's not always true. What we can do instead of having a group of chat messages within a chat is that each chat message is an individual component and we can determine which one is which by getting the sender of that message. If the sender is me, then that is MY chat message which should float to the right.